Chandra/cmd_states/cmd_states.py

```diff
@@ -104,9 +105,22 @@ def add_trans(date=date, **kwargs):
         # if no key in kwargs is in the exclude set then update transition
         if not (exclude and set(exclude).intersection(kwargs)):
             transitions.setdefault(date, {}).update(kwargs)
+            transitions['last_date'] = date
```

Comment:
Not thrilled by the idea of just tossing another key into transitions, but would it be worse to either make a new global-style variable that would need to be updated on every add_trans call?  Or to pass some other data structure to add_trans to allow an update by reference of such a variable?